homeassistant/components/vesync/__init__.py

```diff
@@ -48,6 +51,25 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     hass.data[DOMAIN] = {}
     hass.data[DOMAIN][VS_MANAGER] = manager
 
+    # Create a DataUpdateCoordinator for the manager
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            await hass.async_add_executor_job(manager.update)
+        except Exception as err:
+            raise UpdateFailed(f"Update failed: {err}") from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=DOMAIN,
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=30),
```

Comment:
Could we set this to 60? At 30 it would max the count at 3 devices.   60 will solve that.

Reply:
It should not max out, there should be only one request regardless of the number of devices per second which would be (25 * 3600 / 30) = 2880 per day, right?

The error message mentioned "3200 + 1500 * user owned device number" which was a issue if there were lots of total entities.

Reply:
Ahh let me check the underlying code. I assumed that Vesync library was just doing a for each loop.  The endpoints are different so I believe it is but I'll double check.

Reply:
This [update](https://github.com/webdjoe/pyvesync/blob/c44011abd61b7801f90264cee10676ceb77285a2/src/pyvesync/vesync.py#L418) on the vesync manager updates all the devices within a 30 sec interval check.

Reply:
There will be another 2880 calls per day due to pulling the device list in each `manager.update()` call, which goes beyond the limit. It would be better to use `manager.update_all_devices()` to avoid the unnecessary pulling of the device list. Adding / removing devices can be done manually by a service

Reply:
Oh I see there is call to `get_devices`. That makes sense `update_all_devices` calls `dev.get_details` and the same is called from `update`.

This should get the count back to 2800 + a couple for HA startup. I will try to wrap all this tonight.

Reply:
We can't use `update_all_devices` because that method is incorrectly iterates over `self._dev_list.keys` instead of `self._dev_list.values`. 

I can continue using update for now and then update it one vesync has been fixed?

Reply:
I can't believe I never caught that. 

@joostlek can I make a release that can be bumped in this PR?

Reply:
2.1.15 is published with the fix

Reply:
Thanks for the prompt fix. I rebase my branch and bumped the pevsync version and also selected the `Dependency upgrade` checkbox in the PR.

Reply:
Could we include the bump of the lib in a preliminary PR so it's not included in this (quite large already) PR

Reply:
Sure. I have submitted https://github.com/home-assistant/core/pull/134156 for consuming 2.1.15. I will adjust this PR once that has been merged.

Reply:
On the topic of the calls in the above call it does the following: 

`            for device in chain(*devices):
                device.update()`
https://github.com/webdjoe/pyvesync/blob/c44011abd61b7801f90264cee10676ceb77285a2/src/pyvesync/vesync.py#L435-L436

Which calls once for each device.  Based on this I still believe you will max out calls if not doing 60 second updates.
30 second interval will be: 
1 device = 2880 update requests,  4700 quota
2 devices = 5760 update requests, 6200 quota
3 devices = 8640 update requests, 7700 quota. 

This doesn't account for any room on device list updates, HA restarts etc.     We could go lower than 60 if wanting to consume some of the base 3200 but I was thinking use the "device" added qty and leave the 3200 for restarts, mobile app usage etc.  As such 1500 calls per device is 1.04 calls per minute.

Reply:
I see your point. I think I was more fixated on the issue that each sensor entity was also doing its update. So this PR should still help event for outlet which have energy sensors.

But at device level, 60sec should work. 1500 request works out to be 57.6 seconds.

We can separately consider making this a configuration or dynamically adjusting the next refresh based on device count.

Thank you for your input.

Reply:
I hoped to go dynamic but not aloud by HA. https://github.com/home-assistant/core/pull/127152

Reply:
Bummer ..  then we can adjust the `update_interval` of the coordinator.

Reply:
@gjohansson-ST Would you know if I need to do something else for https://github.com/home-assistant/core/pull/134156 ? Does it need additional approvals?

Reply:
Please rebase this PR to include the latest lib bump. Setting this to draft for now, please click "ready for review" when it is.